packages/dd-trace/src/exporters/agent/request.js

```diff
@@ -44,13 +27,46 @@ function request (options, callback) {
       }
     })
   })
-
   req.setTimeout(options.timeout, req.abort)
-  req.on('error', e => callback(new Error(`Network error trying to reach the agent: ${e.message}`)))
-
   data.forEach(buffer => req.write(buffer))
+  return req
+}
+
+function request (options, callback) {
+  options = Object.assign({
+    headers: {},
+    data: [],
+    timeout: 2000
+  }, options)
+
+  const data = [].concat(options.data)
+  const isSecure = options.protocol === 'https:'
+  const client = isSecure ? https : http
+  const agent = isSecure ? httpsAgent : httpAgent
+
+  options.agent = agent
+  options.headers['Content-Length'] = byteLength(data)
+
+  if (containerId) {
+    options.headers['Datadog-Container-ID'] = containerId
+  }
+  const firstRequest = retriableRequest(options, callback, client, data)
+
+  // The first request will be retried if it fails due to a socket connection close
+  const firstRequestErrorHandler = error => {
+    if (firstRequest.reusedSocket && (error.code === 'ECONNRESET' || error.code === 'EPIPE')) {
+      log.warn('Retrying request due to socket connection error')
```

Comment:
I would say to use `debug` level here since it's not actionable.

Reply:
that makes sense. Done